src/language/variables.md

```diff
@@ -38,6 +38,48 @@ String name = 'Bob';
   of using `var`, rather than type annotations, for local variables.
 {{site.alert.end}}
 
+## Null safety
+
+The Dart language enforces sound null safety.
+
+Null safety prevents an error that results from unintentional access
+of variables set to `null`. The error is called a null dereference error.
+If you try to access a property or call a method on a variable set to `null`,
+that causes a null dereference error.
```

Comment:
Null safety doesn't directly have to do with variables. For example:

```dart
int? someFunction() => null;

main() {
  someFunction().abs();
}
```

Here, there are no variables at all. But this program would throw a null reference exception if Dart didn't have sound null safety.

I would maybe say something like:

A null dereference error occurs when you access a property or call a method on an expression that evaluates to `null` (unless it's a property of method that `null` itself supports, like `toString()` or `hashCode`). With null safety, the Dart compiler detects these potential errors at compile time.

Reply:
Added.